@here/harp-mapview/lib/MapView.ts

```diff
@@ -312,6 +319,48 @@ export interface MapViewOptions {
      */
     theme?: string | Theme | Promise<Theme>;
 
+    // tslint:disable:max-line-length
+    /**
+     * Resolve `URI` referenced in `MapView` assets using this resolver.
+     *
+     * Use, to support application/deployment specific `URI`s into actual `URLs` that can be loaded
+     * with `fetch`.
+     *
+     * As an example, assume that `theme` reference fonts using made-up scheme that resolves files
+     * like [`require.resolve`](https://nodejs.org/api/modules.html#modules_require_resolve_request_options).
+     *
+     * So, assuming theme has this:
+     * ```
+     * "fontCatalogs": [
+     *   {
+     *       "url": "npm://@here/harp-fontcatalog/resources/Default_FontCatalog.json"
+     *   }
+     * ]
+     * ```
+     *
+     * Then,
+     *
+     * 1) you should ensure that `@here/harp-fontcatalog/resources` is either packaged with your app
+     *   or available from some CDN;
+     *
+     * 2) `MapView` should be configured, where to find `@here/harp-fontcatalog` files, for example
+     * with [[createSimpleUrlResolver]]:
+     *
+     * ```
+     * uriResolver: createSimpleUrlResolver({
+     *    // typical, `harp-examples`-like deployment
+     *    "npm://@here/harp-fontcatalog/resources/": "/resources/fonts/",
```

Comment:
That's a nice usecase but I think the main reason why we introduce this is to resolve the asset path like:
`font://FiraSans`, `icon://day/zoo`, ...

Reply:
Documentation updated to be more abstract, see https://github.com/heremaps/harp.gl/pull/901/files